２つセルが表示される用に改修

s-hayashi1104 opened the pull request:
２つのセルが表示されるように改修

s-hayashi1104 committed:
２つセルが表示される用に改修

clonable-eden commented on public/js/components/ReactionTimerGame.js:
```diff
@@ -19,18 +21,31 @@ class ReactionTimerGame {
 
   handleActiveCellSelected() {
+    // クリック回数を保持して一回目はCellRow２回めはCellRow2??
+    // ただそうするとどちらをクリックしてもいいという要件を満たさない気がする
```
そのとおりです。
なので、view側からどこがクリックされたのかを、このhandleActiveCellSelectedに渡してあげる必要があると思いませんか？

clonable-eden reviewed (commented):
考えているコメントに対してコメントしました！

s-hayashi1104 committed:
view側にクリック回数を保持させるよう改修
アクティブセルの２つ目の座標をview側に渡すよう改修

clonable-eden commented on public/js/components/ReactionTimerGridView.js:
```diff
@@ -5,11 +5,12 @@ class ReactionTimerGridView {
   initDomAndListeners() {
     document.querySelector('.btn-start')
-      .addEventListener('click', this.handleBtnStart.bind(this));
+      .addEventListener('click', this.handleBtnStart.bind(this), this.countUpValue += 1);
```
これは何をしているか理解していますか？
https://developer.mozilla.org/ja/docs/Web/API/EventTarget/addEventListener

この呼び出し方だと、
```js
target.addEventListener(type, listener[, useCapture]);
```
となるので、useCaptureに当たる部分に `this.countUpValue += 1`の結果が入るだけです（もちろんthis.countUpValueは1になります）。

clonable-eden replied:
またviewでcountを管理する必要がありますか？gameの方ではダメでしょうか？

clonable-eden commented on public/js/components/ReactionTimerGame.js:
```diff
@@ -19,19 +21,35 @@ class ReactionTimerGame {
+      this.view.countUpValue += 1;
+    } else {
+      this.view.deactivateCell(this.activeCellRow2, this.activeCellCol2);
+    }
```
次にこちら。これだと1回目にクリックされたのがactiveCellRow,activeCellColに該当するセルでなくても、deactivateされてしまいます。
また、再度ゲームを行ったとき、this.view.countUpValueを初期化していないため、常にelseの方に入ってきてしまいます！

clonable-eden reviewed (changes requested):
@s-hayashi1104 コメントしました！

s-hayashi1104 committed:
カウントの回数の保持するclassをview側からgame側に変更
カウントの回数をゲーム終了時に初期化する処理を追加

s-hayashi1104 committed:
■modify
　セルの座標をview側とgame側で保持させる用に改修
■issue
　クリックしていない方のセルが消えることがある
　おそらくクリックの際にきちんと座標が取れていない？？
■question
ReactionTimerGridView.js
私のソースだとsteps 18
  registerActiveCellSelectedCallback(callback) {
    this.callbacks.handleActiveCellSelected = callback;
  }
がいまいち何をしているかわかりません
コンストラクターにbindされているようですが。。。

s-hayashi1104 committed:
■Game.js
・不要なfor文削除
・意味のないif文削除
・コンソールに表示するカウントアップの判定が誤っていたため修正
・２つ目のアクティブなセルをview側に通知できる用に変更

s-hayashi1104 committed:
追加した２つ目のセルを正確に削除できる用に追加したイベントハンドラを
リスナーに渡すよう改修

s-hayashi1104 committed:
view側でセルを保持するのをやめ、コールバックに値を渡せる用に改修
ゲーム側もview側の変更に伴いディアクティブにする際の処理を変更
ハンドラーを一つに改修

clonable-eden commented on public/js/components/ReactionTimerGridView.js:
```diff
@@ -55,17 +57,19 @@ class ReactionTimerGridView {
-  handleActiveCellClick() {
-    this.callbacks.handleActiveCellSelected();
+  handleActiveCellClick(event) {
+    const position = (event.target.id.split(':'));
```
```javascript
const [row, col] = event.target.id.split(':');
```
ってやるとどこかのrepoでやった分割代入！

clonable-eden reviewed (commented):
@s-hayashi1104 コメントしました！機能は大丈夫そうです！

s-hayashi1104 committed:
分割代入に改修

clonable-eden commented on public/js/components/ReactionTimerGridView.js:
```diff
@@ -55,17 +57,17 @@ class ReactionTimerGridView {
-  handleActiveCellClick() {
-    this.callbacks.handleActiveCellSelected();
+  handleActiveCellClick(event) {
+    const [row, col] = (event.target.id.split(':'));
```
👍

clonable-eden reviewed (approved):
@s-hayashi1104 LGTMです！masterにマージしましょう
